metadata-service/auth-api/src/main/java/com/datahub/authentication/Authenticator.java

```diff
@@ -20,7 +20,7 @@ public interface Authenticator {
    * @param authenticatorConfig config provided to the authenticator derived from the Metadata Service YAML config. This
    *                            config comes from the "authentication.authenticators.config" configuration.
    */
-  void init(@Nonnull final Map<String, Object> authenticatorConfig);
+  void init(@Nonnull final Map<String, Object> authenticatorConfig, final Map<String, Object> context);
```

Comment:
We need a followup to change this from a map.. I really hate this map here

Reply:
The ideal end state in my mind: 

Current AuthenticatorContext -> AuthenticationRequest

This map -> AuthenticatorContext

Reply:
Also, if the second argument can be null, you may want to explicitly annoate is as such using @Nullable.

Remember, ANYONE can theoretically write their own Authenticator. This is breaking our public contract and thus backwards incompatible technically. 

We should make sure we get this into the final state that we want it.